tracing/src/main/java/com/palantir/tracing/Tracers.java

```diff
@@ -217,19 +217,24 @@ public static ScheduledExecutorService wrapWithNewTrace(String operation,
         return wrapWithNewTrace(DEFAULT_ROOT_SPAN_OPERATION, delegate);
     }
 
+    public static <V> Callable<V> wrapWithNewTrace(String operation, Callable<V> delegate) {
+        return wrapWithNewTrace(operation, delegate, Optional.empty());
+    }
+
     /**
      * Wraps the given {@link Callable} such that it creates a fresh {@link Trace tracing state} for its execution.
      * That is, the trace during its {@link Callable#call() execution} is entirely separate from the trace at
      * construction or any trace already set on the thread used to execute the callable. Each execution of the callable
      * will have a fresh trace. The given {@link String operation} is used to create the initial span.
      */
-    public static <V> Callable<V> wrapWithNewTrace(String operation, Callable<V> delegate) {
+    public static <V> Callable<V> wrapWithNewTrace(String operation, Callable<V> delegate,
+            Optional<Boolean> isObservable) {
```

Comment:
this seems bonkers

Reply:
looks like the optional boolean is used as a tri-state where the options are: Sample, Don't Sample, Let the sampler decide whether to sample

Reply:
Yeah @iamdanfox if you are referring to the `Optional<Boolean>` then it's already used in the current tracing logic. I can add some comments to explain its purpose since I'm making it part of the public API.

Reply:
Given that we're adding a brand new method here, we have an opportunity to make a better API.   Even with documentation on the method, Optional<Boolean> is a pretty weird looking signature.

I'd propose something like:

```java
   enum Observability {
        SAMPLE,
        DO_NOT_SAMPLE,
        SAMPLER_DECIDES
    }
```

These probably aren't quite the perfect names yet, but I think they're preferrable to Optional<Boolean>!

Reply:
@iamdanfox That makes sense. I initially didn't do this because I didn't want to introduce a breaking change, but agree that it's much clearer with a enum instead of an optional. I opened a separate PR for the refactor: https://github.com/palantir/tracing-java/pull/166